From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
Date: Sat, 4 Mar 2017 00:12:55 +0300
Subject: [ARCHEOLOGY] Remove SPIdev warnings

> X-Git-Archeology: - Revision c43b20256817031186bf78574d589f68cea46779: https://github.com/armbian/build/commit/c43b20256817031186bf78574d589f68cea46779
> X-Git-Archeology:   Date: Sat, 04 Mar 2017 00:12:55 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove SPIdev warnings
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2c08ec8f5a210de35f9482f482ac01ea15381792: https://github.com/armbian/build/commit/2c08ec8f5a210de35f9482f482ac01ea15381792
> X-Git-Archeology:   Date: Thu, 24 May 2018 13:32:29 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: Merge sunxi family into stable
> X-Git-Archeology:
> X-Git-Archeology: - Revision 1a12994e79b6ef173dc58efe4df8919cb6cc7781: https://github.com/armbian/build/commit/1a12994e79b6ef173dc58efe4df8919cb6cc7781
> X-Git-Archeology:   Date: Tue, 17 Jul 2018 15:53:30 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Moving sunxi-next to 4.17.y (#1049)
> X-Git-Archeology:
> X-Git-Archeology: - Revision a57ce78b37f8dd2eb94a3836f4a7f6969f2ffd72: https://github.com/armbian/build/commit/a57ce78b37f8dd2eb94a3836f4a7f6969f2ffd72
> X-Git-Archeology:   Date: Tue, 21 Aug 2018 10:41:10 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Reverting sunxi/sunxi64 NEXT to 4.14. (#1087)
> X-Git-Archeology:
> X-Git-Archeology: - Revision adeb736f8edd153f0144d80a6928dee89b8e9bc6: https://github.com/armbian/build/commit/adeb736f8edd153f0144d80a6928dee89b8e9bc6
> X-Git-Archeology:   Date: Wed, 03 Jul 2019 21:28:48 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: sunxi-dev: fix the same spi bug as in sunxi-next, remove deprecated patches and adjust one
> X-Git-Archeology:
> X-Git-Archeology: - Revision 150ac0c2afa147d9e3b036c8ecd8238fe5648cf3: https://github.com/armbian/build/commit/150ac0c2afa147d9e3b036c8ecd8238fe5648cf3
> X-Git-Archeology:   Date: Tue, 19 Nov 2019 23:25:39 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Remove K<4, change branches, new features (#1586)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 0cdffb29b07305209efb12cf3b5ac6032d3a1153: https://github.com/armbian/build/commit/0cdffb29b07305209efb12cf3b5ac6032d3a1153
> X-Git-Archeology:   Date: Wed, 24 Mar 2021 19:01:53 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Renaming DEV branch to EDGE (#2704)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 990798c3fbfb208b1e333d7e589f01f015e92539: https://github.com/armbian/build/commit/990798c3fbfb208b1e333d7e589f01f015e92539
> X-Git-Archeology:   Date: Sat, 05 Feb 2022 21:53:26 +0100
> X-Git-Archeology:   From: The-going <48602507+The-going@users.noreply.github.com>
> X-Git-Archeology:   Subject: Patching sunxi 5.15 (#3460)
> X-Git-Archeology:
> X-Git-Archeology: - Revision ca478cc4e563655d99fd3380d3b1217481d6da7e: https://github.com/armbian/build/commit/ca478cc4e563655d99fd3380d3b1217481d6da7e
> X-Git-Archeology:   Date: Thu, 12 May 2022 19:11:40 +0200
> X-Git-Archeology:   From: The-going <48602507+The-going@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bugfix spidev (#3737)
> X-Git-Archeology:
> X-Git-Archeology: - Revision e7bd9b8f13af9ee054f44a422b2aca19746b9244: https://github.com/armbian/build/commit/e7bd9b8f13af9ee054f44a422b2aca19746b9244
> X-Git-Archeology:   Date: Sat, 28 May 2022 08:56:19 +0200
> X-Git-Archeology:   From: Tony <tonymckahan@gmail.com>
> X-Git-Archeology:   Subject: Add Spidev workarounds and clean patches (WIP) (#3812)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 73691a9e24440e0a8104b2c25d168ba8947a10ad: https://github.com/armbian/build/commit/73691a9e24440e0a8104b2c25d168ba8947a10ad
> X-Git-Archeology:   Date: Thu, 04 Aug 2022 21:50:40 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: meson64: edge: rework to kernel 5.19 (#3941)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 1b12209ded2c356df514e3dd99bd945c0afd7a32: https://github.com/armbian/build/commit/1b12209ded2c356df514e3dd99bd945c0afd7a32
> X-Git-Archeology:   Date: Tue, 25 Oct 2022 20:38:31 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Bump meson64 edge to 6.0.y (#4341)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5b46bd7273909a2a9688efe85c4d45d00d407865: https://github.com/armbian/build/commit/5b46bd7273909a2a9688efe85c4d45d00d407865
> X-Git-Archeology:   Date: Mon, 12 Dec 2022 08:02:25 +0100
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64-6.0` kernel patches: mbox formatting, archeology to find lost authors/descriptions; rebase against 6.0.12; no actual changes (#4546)
> X-Git-Archeology:
> X-Git-Archeology: - Revision eb7d4a0bd20e56118f9c8c9089c063154c58a239: https://github.com/armbian/build/commit/eb7d4a0bd20e56118f9c8c9089c063154c58a239
> X-Git-Archeology:   Date: Mon, 12 Dec 2022 08:02:49 +0100
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64`: bump `edge` to `6.1-rc8` (#4554)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 321fec6f41fdaa1504755b1a7864415f3c518584: https://github.com/armbian/build/commit/321fec6f41fdaa1504755b1a7864415f3c518584
> X-Git-Archeology:   Date: Mon, 09 Jan 2023 08:22:23 +0100
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: meson64: 6.1.y/edge: rebase patches against v6.1.4 (no actual changes) (#4673)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 80dffbc7611bd76d675fcf74d352e1c55ce51f29: https://github.com/armbian/build/commit/80dffbc7611bd76d675fcf74d352e1c55ce51f29
> X-Git-Archeology:   Date: Tue, 10 Jan 2023 00:31:35 +0100
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64`: `edge`: bump to `6.2` - copy patches as-is from 6.1
> X-Git-Archeology:
> X-Git-Archeology: - Revision 64f9cf9b462c40a946a6a6f802cdb53db4b51b20: https://github.com/armbian/build/commit/64f9cf9b462c40a946a6a6f802cdb53db4b51b20
> X-Git-Archeology:   Date: Sat, 18 Feb 2023 07:46:21 -0300
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64`: edge/6.2: rebase onto `v6.2-rc5`
> X-Git-Archeology:
> X-Git-Archeology: - Revision 755479f40791a7038a3af92b2ccdc2d4b46d06f6: https://github.com/armbian/build/commit/755479f40791a7038a3af92b2ccdc2d4b46d06f6
> X-Git-Archeology:   Date: Tue, 28 Feb 2023 18:54:28 -0300
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: meson64: edge: rebase patches onto `v6.2.1`; fix mbox for recently added patch (no actual changes)
> X-Git-Archeology:
> X-Git-Archeology: - Revision e430fb090f87cae215645866009a6bf53e1f0ddc: https://github.com/armbian/build/commit/e430fb090f87cae215645866009a6bf53e1f0ddc
> X-Git-Archeology:   Date: Tue, 13 Jun 2023 12:33:47 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64` `edge` 6.4: simple fix for drv-spi-spidev-remove-warnings.patch
> X-Git-Archeology:
> X-Git-Archeology: - Revision 38d34e52698eb651866f6586ed4b9e838b76413c: https://github.com/armbian/build/commit/38d34e52698eb651866f6586ed4b9e838b76413c
> X-Git-Archeology:   Date: Tue, 13 Jun 2023 12:33:47 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64` `edge` 6.4: initial copy (as-is) of 6.2 patches
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2230db629924ae0c88de9d8402d5c754d8701c88: https://github.com/armbian/build/commit/2230db629924ae0c88de9d8402d5c754d8701c88
> X-Git-Archeology:   Date: Sat, 02 Sep 2023 14:47:46 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64`/`edge`: bump to 6.5.y; rebase all patches vs `v6.5`(.0)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 3811f36a998aa62e382dfd9cb9284196d9b3a5f0: https://github.com/armbian/build/commit/3811f36a998aa62e382dfd9cb9284196d9b3a5f0
> X-Git-Archeology:   Date: Tue, 03 Oct 2023 13:53:40 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `meson64`/`edge`: bump to `6.6-rc4`; initial copy from 6.5
> X-Git-Archeology:
> X-Git-Archeology: - Revision 1ffdab4fe47413fca5438afe2c98828c6bb955ea: https://github.com/armbian/build/commit/1ffdab4fe47413fca5438afe2c98828c6bb955ea
> X-Git-Archeology:   Date: Wed, 14 Aug 2024 17:39:38 -0400
> X-Git-Archeology:   From: Alessandro Lannocca <alessandro.lannocca@gmail.com>
> X-Git-Archeology:   Subject: meson64: current: refresh drv-spi-spidev-remove-warnings.patch
> X-Git-Archeology:
> X-Git-Archeology: - Revision a53be7e6220ff365c3d745e4fac1c79ea5fd12d4: https://github.com/armbian/build/commit/a53be7e6220ff365c3d745e4fac1c79ea5fd12d4
> X-Git-Archeology:   Date: Tue, 01 Oct 2024 14:17:50 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor@armbian.com>
> X-Git-Archeology:   Subject: Adjust patches due to kernel bump
> X-Git-Archeology:
---
 drivers/spi/spidev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 111111111111..222222222222 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -704,6 +704,7 @@ static const struct file_operations spidev_fops = {
 static struct class *spidev_class;
 
 static const struct spi_device_id spidev_spi_ids[] = {
+	{ .name = "spi-dev" },
 	{ .name = "bh2228fv" },
 	{ .name = "dh2228fv" },
 	{ .name = "jg10309-01" },
@@ -734,6 +735,7 @@ static int spidev_of_check(struct device *dev)
 }
 
 static const struct of_device_id spidev_dt_ids[] = {
+	{ .compatible = "armbian,spi-dev", .data = &spidev_of_check },
 	{ .compatible = "cisco,spi-petra", .data = &spidev_of_check },
 	{ .compatible = "dh,dhcom-board", .data = &spidev_of_check },
 	{ .compatible = "elgin,jg10309-01", .data = &spidev_of_check },
-- 
Armbian

